src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1507,13 +1507,16 @@ public bool OpenCustomNodeWorkspace(Guid guid)
         /// </summary>
         /// <param name="node"></param>
         /// <param name="centered"></param>
-        public void AddNodeToCurrentWorkspace(NodeModel node, bool centered)
+        public void AddNodeToCurrentWorkspace(NodeModel node, bool centered, bool addToSelection = true)
         {
             CurrentWorkspace.AddAndRegisterNode(node, centered);
 
             //TODO(Steve): This should be moved to WorkspaceModel.AddNode when all workspaces have their own selection -- MAGN-5707
-            DynamoSelection.Instance.ClearSelection();
-            DynamoSelection.Instance.Selection.Add(node);
+            if (addToSelection)
```

Comment:
Need to be able to create and add a new node automatically but with the option of not switching selection from existing nodes.